### PR TITLE
Lianghy0326-proposal-submission

### PR DESCRIPTION
**Pull request by Lianghy0326**
---
A good PR should include the following parts:

- [x] Provide a clear title
- [x] Provide meaningful descriptions in the PR and the commit logs
- [x] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [x] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [x] Exclude irrelevant files, such as binaries, text files, or dot files
- [x] Make sure tests/CI pass

### CODE REVIEW

File: project/Lianghy0326/Readme.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
+=============================
+Path Finding Using Algorithms
+=============================
+
+Basic Information
+=================
+
+Github Repository: https://github.com/Lianghy0326/Path-finding-visualization
+
+Pathfinding is a fundamental problem in computer science and has numerous applications in various domains, including robotics, video games, and network routing.
```
Comment (yungyuc) [inline project/Lianghy0326/Readme.rst]: While it is fundamental, it is not so interesting for you to easily make a presentation.  If you are interested in computer vision or graphics, a topic in those areas may be easier to present.

If you are not 100% sure about what you want to do, please find me at [sciwork #nsd channel](https://discord.gg/tZsUnx4XmB) to discuss more.

File: project/Lianghy0326/Readme.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+=============================
+
+Basic Information
+=================
+
+Github Repository: https://github.com/Lianghy0326/Path-finding-visualization
+
+Pathfinding is a fundamental problem in computer science and has numerous applications in various domains, including robotics, video games, and network routing. 
+The objective of pathfinding algorithms is to find the optimal path between two points in a given environment while considering obstacles, costs, and other constraints.
+In this project, we aim to explore and implement different pathfinding algorithms, such as A* (A-star), Depth-First Search (DFS), and possibly others, to address the diverse needs of users in path planning scenarios. Through this endeavor, we seek to provide researchers and developers with a versatile toolkit for efficiently navigating complex environments and finding optimal paths. Additionally, we will develop visualization tools to aid in understanding the behavior and performance of these algorithms.
```
Comment (yungyuc) [inline project/Lianghy0326/Readme.rst]: It is bad to use one line for a single paragraph, for it makes the lines to be too long. It is hard to add comments at specific location for a long line. Please change to 79-char width.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [ ] You missed an entry in `project/README.rst` as [requested](https://github.com/yungyuc/nsdhw_24sp/blob/567be93bc63a6fb1c2c55b8faf5825172daeaba5/project/README.rst?plain=1#L22)
- [x] Discuss ease of presentation
- [x] 79-char line width
- [ ] Test earlier for productivity

When you are ready for the next review, leave a comment, clearly write down something like "I am ready, please review again".

**Comment by yungyuc** [comment]
@Lianghy0326 Please update the proposal.

**Comment by Lianghy0326** [comment]
Already update the proposal project.
I have changed the proposal topic from "path-finding" to "ODE solver" topic.
thanks teacher.

Lianghy0326

Yung-Yu Chen ***@***.***> 於 2024年4月14日 週日 下午9:08寫道：

File: project/Lianghy0326/Readme.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# DE
+Automatic Differential Equation Solver Optimizer
+
+
+## Basic Information
+Github Repository: https://github.com/Lianghy0326/DE
+
+
+Design an intelligent C++ solver that automatically selects the optimal solving method for differential equation (DE) problems.
```
Comment (yungyuc) [inline project/Lianghy0326/Readme.md]: This is way longer than 79 chars.  You should configure your editor to format your text.

You have most of the lines to be short enough for review, so it's not too bad.  Please fix the line width when you update for other things.

File: project/Lianghy0326/Readme.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# DE
```
Comment (yungyuc) [inline project/Lianghy0326/Readme.md]: I like short project name 🙂

File: project/Lianghy0326/Readme.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+* ``gtest``: Used for unit testing.
+
+## Schedule
+- Week 1 (04/08): Research
+- Week 2 (04/15): Research
+- Week 3 (04/22): Research && Implementation
+- Week 4 (04/29): Implementation
+- Week 5 (05/06): Implementation
+- Week 6 (05/13): Implementation && Test and debug
+- Week 7 (05/20): Test and debug
```
Comment (yungyuc) [inline project/Lianghy0326/Readme.md]: Still so late for testing.  It's dangerous for your progress.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [ ] (Again) You missed an entry in `project/README.rst` as [requested](https://github.com/yungyuc/nsdhw_24sp/blob/567be93bc63a6fb1c2c55b8faf5825172daeaba5/project/README.rst?plain=1#L22)
- [ ] Please find me in scisprint or meetup or schedule discussions for your project.  It's not in a good shape and you need help of critical analysis.

When you are ready for the next review, leave a comment, clearly write down something like "I am ready, please review again".  Do not forget this when you participate in a code review (and/or PR).

**Comment by yungyuc** [comment]
> When you are ready for the next review, leave a comment, clearly write down something like "I am ready, please review again". Do not forget this when you participate in a code review (and/or PR).

@Lianghy0326 I have not heard from you.  Please address the points I made or I cannot grade.

**Comment by Lianghy0326** [comment]
I am ready, please review again
I have fixed my readme.rst file in my project

**Comment by Lianghy0326** [comment]
I am ready, please review again

File: hw4/.gitignore
Comment (yungyuc) [inline hw4/.gitignore:1]: Your proposal PR should not include anything else.  Please rebase to fix the wrong history.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [ ] (Again twice) You missed an entry in `project/README.rst` as [requested](https://github.com/yungyuc/nsdhw_24sp/blob/567be93bc63a6fb1c2c55b8faf5825172daeaba5/project/README.rst?plain=1#L22)
- [ ] I did not hear from you anywhere except the late comments in this PR.  Please find me to discuss your progress.
- [ ] Your PR includes unrelated history.  Please get rid of it.

**Comment by Lianghy0326** [comment]
I'm unsure how to resolve the issue with the "missed entry in project/README.rst."

Additionally, I have made modifications to my project content to address the topic's shortcomings and would appreciate critical analysis. If the revisions are satisfactory, I would like to discuss my topic next week after class or other time.

Thank you teacher.

ViviLiang

**Comment by yungyuc** [comment]
Please read https://github.com/yungyuc/nsdhw_24sp/blob/567be93bc63a6fb1c2c55b8faf5825172daeaba5/project/README.rst?plain=1#L22 as I mentioned in the previous comments.

I think it's a good idea to have a F2F discussion.  We can do it during the lecture.

**Pull request merged**